dashboard/modules/actor/actor_head.py

```diff
@@ -0,0 +1,206 @@
+                            node_actors.setdefault(
+                                node_id, {})[actor_id] = actor_table_data
+                    DataSource.job_actors.reset(job_actors)
+                    DataSource.node_actors.reset(node_actors)
+                    logger.info("Received %d actor info from GCS.",
+                                len(actors))
+                    break
+                else:
+                    raise Exception(
+                        f"Failed to GetAllActorInfo: {reply.status.message}")
+            except Exception:
+                logger.exception("Error Getting all actor info from GCS.")
+                await asyncio.sleep(stats_collector_consts.
+                                    RETRY_GET_ALL_ACTOR_INFO_INTERVAL_SECONDS)
+
+        # Receive actors from channel.
+        state_keys = ("state", "address", "numRestarts", "timestamp", "pid")
+        async for sender, msg in receiver.iter():
+            try:
+                actor_id, actor_table_data = msg
+                pubsub_message = ray.gcs_utils.PubSubMessage.FromString(
+                    actor_table_data)
+                message = ray.gcs_utils.ActorTableData.FromString(
+                    pubsub_message.data)
+                actor_table_data = actor_table_data_to_dict(message)
+                _process_actor_table_data(actor_table_data)
+                # If actor is not new registered but updated, we only update
+                # states related fields.
+                if actor_table_data["state"] != "DEPENDENCIES_UNREADY":
+                    actor_id = actor_id.decode("UTF-8")[len(
+                        ray.gcs_utils.TablePrefix_ACTOR_string + ":"):]
+                    actor_table_data_copy = dict(DataSource.actors[actor_id])
+                    for k in state_keys:
+                        actor_table_data_copy[k] = actor_table_data[k]
+                    actor_table_data = actor_table_data_copy
+                actor_id = actor_table_data["actorId"]
+                job_id = actor_table_data["jobId"]
+                node_id = actor_table_data["address"]["rayletId"]
+                # Update actors.
+                DataSource.actors[actor_id] = actor_table_data
+                # Update node actors (only when node_id is not Nil).
+                if node_id != stats_collector_consts.NIL_NODE_ID:
+                    node_actors = dict(DataSource.node_actors.get(node_id, {}))
+                    node_actors[actor_id] = actor_table_data
+                    DataSource.node_actors[node_id] = node_actors
+                # Update job actors.
+                job_actors = dict(DataSource.job_actors.get(job_id, {}))
+                job_actors[actor_id] = actor_table_data
+                DataSource.job_actors[job_id] = job_actors
+            except Exception:
+                logger.exception("Error receiving actor info.")
+
+    @routes.get("/logical/actor_groups")
```

Comment:
Maybe renaming it to `all_actors`?

Reply:
This API is used by the frontend, I just moved it from the logical_view module. Do we really need to rename it to `all_actors`?